lesson05/main.js

```diff
@@ -1,17 +1,22 @@
-const ul = document.getElementById("js-ul"); 
-const fragment = document.createDocumentFragment(); 
 const elements = [ 
   {to: "bookmark.html", img:"img/1.png", alt:"画像1", text: "ブックマーク"}, 
   {to: "message.html", img:"img/2.png", alt:"画像2", text: "メッセージ"} 
 ]; 
-elements.forEach(function(element){ 
-  const li = document.createElement("li"); 
-  const a = document.createElement("a"); 
-  const img = document.createElement("img"); 
-  a.href = element.to; 
-  a.textContent = element.text; 
-  img.src = element.img; 
-  img.alt = element.alt; 
-  fragment.appendChild(li).appendChild(a).insertAdjacentElement("afterbegin",img); 
-}); 
-ul.appendChild(fragment);
+
+new Promise((resolve)=> {
+  resolve(elements);
+}).then((elements)=> {
+  const ul = document.getElementById("js-ul"); 
+  const fragment = document.createDocumentFragment(); 
+  elements.forEach((element)=> { 

```

Comment:
IMO 

You can use for...of method instead of forEach method. I was advised that for.. of method can perform better than forEach method. 

[Article](https://auto-worker.com/blog/?p=5715)
I suppose it would be more of your preference to use forEach or for...of in this lesson, but according to the article attached, for..of method has some advantages that forEach does not have. For example, you can use break or continue to stop the process. 

```suggestion 
for( const element of elements){
  const li = document.createElement("li"); 
    const a = document.createElement("a"); 
    const img = document.createElement("img"); 
    a.href = element.to; 
    a.textContent = element.text; 
    img.src = element.img; 
    img.alt = element.alt; 
    fragment.appendChild(li).appendChild(a).insertAdjacentElement("afterbegin",img); 
}
```

Reply:
Thank you saki-san :）
I read the article.
you help me every time :)